### DISCUSSION TITLE
Allow all custom fee collectors to be exempt

### DISCUSSION THREAD

**Discussion by tinker-michaelj** [hiero-ledger/hiero-improvement-proposals#571] [Ideas]
[HIP-18 (Custom Hedera Token Service Fees)](https://hips.hedera.com/hip/hip-18) did not stipulate any exemptions for custom fees. The [final implementation spec](https://github.com/hashgraph/hedera-services/blob/master/docs/fees/custom-fees-characterization.md) carved out two exemptions,

> There are two exemptions from the charging policy. First, token treasury accounts are never charged custom fees. Second, a fee collector account is not charged any fractional fees for which it is the collector.

So the network still charges a fee collector custom fees for which it is **not** the collector. This can be cumbersome for  fungible tokens with multiple fee collectors who exchange units of the token.

We propose to add a field to the [`CustomFee` protobuf message](https://github.com/hashgraph/hedera-protobufs/blob/main/services/custom_fees.proto#L101) so a custom fee can be created with unconditional exemptions for all fee collectors in the owning token type.
```
message CustomFee {
  ...
  AccountID fee_collector_account_id = 3;
  bool all_collectors_are_exempt = 4;
}
```

By default `exempt_all_collectors = false`, and the current implementation's behavior will be preserved. But suppose token type `0.0.T` has two custom fees `F1` and `F2` with respective collectors `0.0.A` and `0.0.B`. If `F1` was created with `all_collectors_are_exempt = true`, then when `0.0.B` sends units of `0.0.T` to `0.0.A`, it will **not** pay fee `F1`.

**Comment by Cooper-Kunz** [comment]
This makes sense @tinker-michaelj - can anyone / any key update this value?

**Reply by tinker-michaelj** [reply]
Hi @Cooper-Kunz, sorry I missed this! 

The fee schedule must be present to update this; and it must sign the `TokenFeeScheduleUpdate`.
